File: file-management-chaincode/models/file.go
package models

type File struct {
	ID        string `json:"id"`
	Name      string `json:"name"`
	Hash      string `json:"hash"`
	Timestamp string `json:"timestamp"`
	Owner     string `json:"owner"`
	Metadata  string `json:"metadata"`
	// AccessList []string `json:"accessList"`
}

// type Access struct {
// 	UserID      string `json:"userID"`
// 	AccessLevel string `json:"accessLevel"`
// }

File: web/src/types/file.ts
export interface File {
    id: string;
    name: string;
    hash: string;
    timestamp: string;
    owner: string;
    metadata: string;
    version: number;
    previousID?: string;
    ipfsLocation: string;
    status: string;               // "PENDING" or "APPROVED"
    requiredOrgs: string[];      // List of required MSP IDs
    currentApprovals: string[];  // List of MSP IDs that have approved
    endorsementType: string;     // "ANY_ORG", "ALL_ORGS", "SPECIFIC_ORGS"
  }